src/pool/pool_contract.ts

```diff
@@ -148,79 +157,187 @@ export abstract class PoolContract extends Contract {
+   * @param contractArgs.spender - The address of the user who is sending tokens to the pool.
+   * @param contractArgs.to - The address of the user who is receiving tokens from the pool.
+   * @param contractArgs.requests - A list of requests to be processed.
+   *
+   * @throws Will throw an error if the request cannot be completed due to insufficient funds or invalid health factor.
+   *
+   * @returns A base64-encoded string representing the operation.
+   */
   submit(contractArgs: SubmitArgs): string {
     return this.call('submit', ...PoolContract.spec.funcArgsToScVals('submit', contractArgs)).toXDR(
       'base64'
     );
   }
 
+  /**
+   * Manages bad debt. Debt is considered "bad" if there is no longer any collateral posted.
+   *
+   * @param user - The user who currently possesses bad debt.
+   *
+   * @throws Will throw an error if the user has collateral posted.
+   *
+   * @returns A base64-encoded string representing the operation.
+   */
   badDebt(user: Address | string): string {
     return this.call('bad_debt', ...PoolContract.spec.funcArgsToScVals('bad_debt', { user })).toXDR(
       'base64'
     );
   }
 
+  /**
+   * Updates the pool status based on the backstop state.
+   *
+   * @throws Will throw an error if the pool is currently on status 4, "admin-freeze",
+   * where only the admin can perform a status update via `setStatus`.
+   *
+   * @returns A base64-encoded string representing the operation.
+   */
   updateStatus(): string {
     return this.call(
       'update_status',
       ...PoolContract.spec.funcArgsToScVals('update_status', {})
     ).toXDR('base64');
   }
 
+  /**
+   * Sets the pool status. Only callable by the admin.
+   *
+   * @param pool_status - The pool status to be set: 0 (active), 2 (on-ice), or 4 (frozen).
+   *
+   * @throws Will throw an error if the caller is not the admin or if the specified conditions are not met for the status to be set.
+   *
+   * @returns A base64-encoded string representing the operation.
+   */
   setStatus(pool_status: u32): string {
     return this.call(
       'set_status',
       ...PoolContract.spec.funcArgsToScVals('set_status', { pool_status })
     ).toXDR('base64');
   }
 
+  /**
+   * Consumes emissions from the backstop and distributes to the reserves based on the reserve emission configuration.
+   *
+   * @returns A base64-encoded string representing the operation.
+   */
   gulpEmissions(): string {
     return this.call(
       'gulp_emissions',
       ...PoolContract.spec.funcArgsToScVals('gulp_emissions', {})
     ).toXDR('base64');
   }
 
+  /**
+   * Sets the emission configuration for the pool. Only callable by the admin.
+   *
+   * @param res_emission_metadata - A vector of ReserveEmissionMetadata to update metadata to.
+   *
+   * @throws Will throw an error if the caller is not the admin or if the sum of ReserveEmissionMetadata shares is greater than 1.
```

Comment:
this is not true for V2.

Better to omit IMO since not a lot of new pools will be being created in v1 anymore